shared/javascripts/host_page_integration.js

```diff
@@ -32,8 +32,69 @@ var privlyHostPage = {
         element.setAttribute("frame_id", frameId);  
         document.documentElement.appendChild(element);    
         element.dispatchEvent(evt);
-        // Send the message "id,height" to the parent window
-        parent.postMessage(frameId+","+height,"*");
+        // Send the id and height to the parent window as JSON object with resize as the command.
+        var resizeMsg = { 'command':'resize', 'frameID':frameId, 'heightNew':height};
+        parent.postMessage(JSON.stringify(resizeMsg),"*");
+        element.parentNode.removeChild(element);
+    },
+
+    /**
+     * Send the parent document the name of this iframe. This will
+     * allow the host page to hide the iframe and show the original link
+     * instead. 
+     *
+     */
+    hideWrapper: function(){
+        var evt = document.createEvent("Events");  
+        evt.initEvent("IframeRemoveEvent", true, false);
+        var element = document.createElement("privElement");
+        var frameId = window.name;
```

Comment:
These functions are largely duplicated 3 times. Maybe one helper function that is called with three different params?